vm-virtio/src/vhost_user/fs.rs

```diff
@@ -144,6 +147,66 @@ impl VhostUserMasterReqHandler for SlaveReqHandler {
 
         Ok(())
     }
+
+    fn fs_slave_io(&mut self, fs: &VhostUserFSSlaveMsg, fd: RawFd) -> HandlerResult<()> {
+        debug!("fs_slave_io");
+
+        for i in 0..VHOST_USER_FS_SLAVE_ENTRIES {
+            // Ignore if the length is 0.
+            if fs.len[i] == 0 {
+                continue;
+            }
+
+            let mut foffset = fs.fd_offset[i];
+            let mut len = fs.len[i] as usize;
+            let gpa = fs.cache_offset[i];
+            if gpa < self.cache_offset.raw_value()
+                || gpa >= self.cache_offset.raw_value() + self.cache_size
+            {
+                return Err(io::Error::new(
+                    io::ErrorKind::Other,
+                    "gpa is out of cache range",
+                ));
+            }
+
+            let offset = gpa.checked_sub(self.cache_offset.raw_value()).unwrap();
```

Comment:
Could you please do a `map_err()` instead of `unwrap()`?

Reply:
OK. the checked_sub should always be Some(), because I've checked both (gpa < cache_offset) and (gpa >= cache_offset + cache_size) above. I was wondering if there is a helper similar to checked_sub checking both branches